lib/pages/timetable/service/uni_event_provider.dart

```diff
@@ -359,6 +359,24 @@ class UniEventProvider extends EventProvider<UniEventInstance>
         .first;
   }
 
+  Future<Iterable<UniEvent>> getAllClassesEvents(String classId) async {
+    return _events
+        .map((events) =>
+            events.where((event) => event.classHeader.id == classId))
+        .first;
+  }
+
+  Future<Iterable<UniEventInstance>> getAllInstancesOfEvent(
```

Comment:
This method is unnecessary. You can just call `event.generateInstances()` directly.